test/e2e_env/universal/auth/dp_auth.go

```diff
@@ -71,4 +74,58 @@ networking:
 			return env.Cluster.GetKumactlOptions().RunKumactlAndGetOutput("get", "dataplanes", "-oyaml")
 		}, "30s", "1s").ShouldNot(ContainSubstring("192.168.0.2"))
 	})
+
+	It("should revoke token and kick out dataplane proxy out of the mesh", func() {
+		// given
+		serviceName := "test-server-to-be-revoked"
+		token, err := env.Cluster.GetKuma().GenerateDpToken(meshName, serviceName)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = env.Cluster.Install(TestServerUniversal(serviceName, meshName, WithServiceName(serviceName), WithToken(token)))
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			online, found, err := IsDataplaneOnline(env.Cluster, meshName, serviceName)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(found).To(BeTrue())
+			g.Expect(online).To(BeTrue())
+		}).Should(Succeed())
+
+		// when token ID is added to revocation list
+		claims := &jwt.RegisteredClaims{}
+		_, _, err = jwt.NewParser().ParseUnverified(token, claims)
+		Expect(err).ToNot(HaveOccurred())
+
+		yaml := fmt.Sprintf(`
+type: Secret
+mesh: dp-auth
+name: dataplane-token-revocations-dp-auth
+data: %s`, base64.StdEncoding.EncodeToString([]byte(claims.ID)))
+		Expect(env.Cluster.Install(YamlUniversal(yaml))).To(Succeed())
+
+		// then DPP is disconnected
+		Eventually(func(g Gomega) {
+			// we need to trigger XDS config change for this DP to disconnect it
+			// this limitation may be lifted in the future
+			yaml = fmt.Sprintf(`
+type: Retry
+name: retry-policy
+mesh: dp-auth
+sources:
+- match:
+    kuma.io/service: test-server-to-be-revoked
+destinations:
+- match:
+    kuma.io/service: test-server-to-be-revoked
+conf:
+  http:
+    numRetries: %d
+`, rand.Int()%100)
+			g.Expect(env.Cluster.Install(YamlUniversal(yaml))).To(Succeed())
```

Comment:
Just trying to understand the behavior, this XDS change has to be retried every time?

Reply:
We need at least one DiscoveryRequest from Envoy to re-authenticate and drop the connection.

I could do the steps
1) Apply revocation config
2) Apply RateLimit to trigger DiscoveryRequest
3) Eventually DP is offline

The problem is that auth uses cached ReadOnlyResourceManager so we may hit a cache with the old revocation list if we only trigger XDS change once. That's why I'm triggering it within the Eventually.